packages/shell-api/src/collection.ts

```diff
@@ -641,12 +641,13 @@ export default class Collection extends ShellApiClass {
   @serverVersions(['3.2.0', ServerVersions.latest])
   async insertMany(docs: Document[], options: BulkWriteOptions = {}): Promise<InsertManyResult> {
     assertArgsDefined(docs);
+    const d: Document[] = Array.isArray(docs) ? docs.map((doc) => ({ ...doc })) : docs;
```

Comment:
Isnt `: docs` also `[{ ...docs }]` ?

Reply:
Not really. I case of `insertMany` we expect only array as an input parameter, and if this is not an array but something else we want to throw the error that the docs parameter must be an array of documents, so we don't want to manually wrap it into the array.

Reply:
And it would not inject the _id if i pass an array?

Reply:
Oh sorry I got confused, you may want to write this in a bit more clear way: If we are assuming docs is already an array, then you can just do this:

``` js
const clonedDocs = docs.map(doc => {...doc});
```

Reply:
this way is clear the intent, is hard to me to understand that we check fort a non legit condition and then we don't do anything about it, there is a lot of implicit going on.

Reply:
But in this case, if the user will pass not an array but an object for example the error will be that the map is not a function instead of server's error that parameter must be an array.

Reply:
So the idea here is that if the input parameter is an array, clone the input array to prevent mutation. If the input parameter is anything else but an array, pass the original value to the driver, it will throw the proper error and won't mutate the parameter because the command won't succeed.